apps/consulting/pages/_document.tsx

```diff
@@ -1,19 +1,45 @@
 import { FC } from 'react';
 
 import { Html, Head, Main, NextScript } from 'next/document';
+import Script from 'next/script';
 
 const Document: FC = () => {
   return (
     <Html lang="en" className="text-[62.5%]">
       <Head>
+        {/*
+          This script collects any GCLID and UTM query parameters from incoming
+          traffic and stores them in `sessionStorage`. It uses a 'quansight_'
+          prefix on the names of all stored parameters to reduce the possibility
+          of collision with other sites that might be using sessionStorage for
+          similar purposes.
+
+          The GCLID capture is disabled for now, since we're not sure how
+          extensively we will be running Google Ads, and it simplifies any
+          privacy questions to omit its capture.
+
+          Search params approach from https://stackoverflow.com/a/901144/4376000.
+        */}
+        <Script id="session-store-url-params" strategy="beforeInteractive">
```

Comment:
Not that it probably makes any discernible difference, but is `beforeInteractive` really the right value for strategy here?

Reply:
I debated about that. `afterInteractive` should be fine, too? It should be a super-quick-to-run script, so I figured pushing it as early as possible would be fine, not slow pageload much. Totally fine to switch it, though.

Reply:
I'm fine either way. Just thought it was worth asking about and perhaps also commenting why which strategy was chosen.